crates/vision/src/robot_detection.rs

```diff
@@ -1,332 +1,236 @@
-) -> Vec<ScoredCluster> {
-    clusters
-        .into_iter()
-        .filter(|cluster| cluster.score >= minimum_cluster_score)
-        .collect()
+fn boxes_from_output(
+    values: &[f32; NUMBER_OF_SCALINGS * PARAMETERS_PER_BOX],
+    grid_position: Vector2<f32>,
+    grid_size: Vector2<f32>,
+    camera_image_size: Vector2<f32>,
+    box_scalings: &[Vector2<f32>; 4],
+) -> [Box; NUMBER_OF_SCALINGS] {
+    let values = values.map(map_to_zero_one);
+    [
+        box_from_network_data(
+            Vector2::new(values[0], values[1]),
+            Vector2::new(values[2], values[3]),
+            values[4],
+            values[5],
+            grid_position,
+            grid_size,
+            camera_image_size,
+            box_scalings[0],
+        ),
+        box_from_network_data(
+            Vector2::new(values[6], values[7]),
+            Vector2::new(values[8], values[9]),
+            values[10],
+            values[11],
+            grid_position,
+            grid_size,
+            camera_image_size,
+            box_scalings[1],
+        ),
+        box_from_network_data(
+            Vector2::new(values[12], values[13]),
+            Vector2::new(values[14], values[15]),
+            values[16],
+            values[17],
+            grid_position,
+            grid_size,
+            camera_image_size,
+            box_scalings[2],
+        ),
+        box_from_network_data(
+            Vector2::new(values[18], values[19]),
+            Vector2::new(values[20], values[21]),
+            values[22],
+            values[23],
+            grid_position,
+            grid_size,
+            camera_image_size,
+            box_scalings[3],
+        ),
+    ]
 }
 
-fn filter_clusters_via_cones(
-    clusters: Vec<ScoredCluster>,
-    cluster_cone_radius: f32,
-) -> (Vec<ScoredCluster>, Vec<ClusterCone>) {
-    clusters
-        .into_iter()
-        .fold((vec![], vec![]), |(mut clusters, mut cones), cluster| {
-            let cone = ClusterCone::from_cluster(&cluster, cluster_cone_radius);
-            if !cones
-                .iter()
-                .any(|existing_cone| existing_cone.intersects_with(&cone))
-            {
-                clusters.push(cluster);
-                cones.push(cone);
-            }
-            (clusters, cones)
-        })
+#[allow(clippy::too_many_arguments)]
+fn box_from_network_data(
+    center: Vector2<f32>,
+    size: Vector2<f32>,
+    probability: f32,
+    distance: f32,
+    grid_position: Vector2<f32>,
+    grid_size: Vector2<f32>,
+    camera_image_size: Vector2<f32>,
+    scaling: Vector2<f32>,
+) -> Box {
+    Box {
+        center: (center + grid_position)
+            .component_div(&grid_size)
+            .component_mul(&camera_image_size)
+            .into(),
+        size: (size * 10.0)
+            .component_mul(&scaling)
+            .component_mul(&camera_image_size)
+            .component_div(&grid_size),
+        probability,
+        distance: distance * 10.0,
```

Comment:
Where does this factor (10) come from? Is it always the same for both cases? It may be a good idea to create a constant for this.

Reply:
haha, I actually do not remember that myself :D ... i have to look it up again. I should have given it a name in the first place... @jonathan-hellwig do you still remember what this was?

Reply:
I think we scaled it by 10 because B-Human does so as well. I think it is just a conversion of units.

Reply:
done